assets/js/base/components/product-grid-item/index.js

```diff
@@ -15,37 +15,56 @@ import ProductImage from './product-image';
 import ProductRating from './product-rating';
 import ProductSaleBadge from './product-sale-badge';
 
-const ProductGridItem = ( { attributes, product = {} } ) => {
+const ProductGridItem = ( { attributes, product } ) => {
 	const { contentVisibility } = attributes;
 	const { button, price, rating, title, image } = contentVisibility;
 	const isLoading = ! Object.keys( product ).length > 0;
-	const classes = classNames(
-		'wc-block-grid__product',
-		{
-			'is-loading': isLoading,
-		},
-	);
+	const classes = classNames( 'wc-block-grid__product', {
+		'is-loading': isLoading,
+	} );
+
 	return (
 		<li className={ classes } aria-hidden={ isLoading }>
-			<ProductLink product={ product } className="wc-block-grid__product-link">
+			<ProductLink
+				permalink={ product.permalink }
+				className="wc-block-grid__product-link"
+			>
 				{ image && (
-					<ProductImage product={ product } className="wc-block-grid__product-image" />
+					<ProductImage
+						images={ product.images }
+						className="wc-block-grid__product-image"
+					/>
 				) }
 				{ title && (
-					<ProductTitle product={ product } className="wc-block-grid__product-title" />
+					<ProductTitle
+						name={ product.name }
+						className="wc-block-grid__product-title"
+					/>
+				) }
+				{ price && (
+					<ProductSaleBadge
+						onsale={ product.onsale }
+						className="wc-block-grid__product-onsale"
+					/>
 				) }
 			</ProductLink>
 			{ price && (
-				<ProductSaleBadge product={ product } className="wc-block-grid__product-onsale" />
-			) }
-			{ price && (
-				<ProductPrice product={ product } className="wc-block-grid__product-price" />
+				<ProductPrice
+					prices={ product.prices }
```

Comment:
Just curious why this change was made? I understand some of these components may only use a small subset of data, but passing the full product object would let them do more in the future, for example, if we needed to introduce some product type specific logic.

Reply:
It's usually a good practice to pass as little as possible in component props, that might avoid useless re-renders and makes it easier to manage default props, for example.

If any of these components needs more `product` properties in the future, I think it's quite easy to add them in another prop. Does that sound good?